buffer/src/main/java/io/netty/buffer/AbstractByteBuf.java

```diff
@@ -1376,6 +1376,10 @@ protected final void checkIndex(int index, int fieldLength) {
 
     private static void checkRangeBounds(final int index, final int fieldLength, final int capacity) {
         if (isOutOfBounds(index, fieldLength, capacity)) {
+            if (capacity == -1) {
```

Comment:
This check is required only for `PooledSlicedByteBuf` which performs a bounds check involving `maxCapacity` prior to accessibility check (and so would throw `IndexOutOfBoundsException` rather than `IllegalReferenceCountException` in the inaccessible case otherwise - failing some unit tests).